test/web-client-test.js

```diff
@@ -0,0 +1,33 @@
+const { assert } = require('chai');
+const sinon = require('sinon');
+const rewiremock = require('rewiremock/node');
+
+/* eslint-disable-next-line global-require */
+rewiremock(() => require('@slack/web-api')).with({
+  WebClient: class {
+    constructor(token, options) {
+      this.token = token;
+      this.options = options || {};
+    }
+  },
+});
+rewiremock.enable();
+const { createWebClient } = require('../src/web-client');
+
+rewiremock.disable();
+
+describe('web-client', () => {
+  beforeEach(() => {
+    sinon.reset();
+  });
+
+  it('should create WebClient with an https proxy agent if given a proxy URL', async () => {
+    const web = createWebClient('xoxb-xxxxx', 'http://test.proxy:8080/');
+    assert.equal(typeof web.options.agent, 'object');
+  });
+
+  it('should create WebClient with default settings if not given a proxy URL', async () => {
+    const web = createWebClient('xoxb-xxxxx');
```

Comment:
thoughts on adding an assertion here to make sure it is creating an object of type WebClient with expected properties? this can also be added to the first test as well!

Reply:
If I understand correctly, the mock on line 6-17 replaces the normal WebClient class which means there is not much left to test. The alternative approach would be to not mock the WebClient and verify that object instead, but it's a pretty complex object and the proxy agent isn't even configured in the web client directly, but instead in the Axios client that it creates internally. I tried going that way at first, but it felt too much like digging around in implementation details of the web-api package.